File: DXDAL/Locations.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Namespace SPGData
    Public Class Locations
        Inherits AuditableBase

        Dim fLocationCode As String
        <Size(25)>
        Public Property LocationCode() As String
            Get
                Return fLocationCode
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationCode", fLocationCode, value)
            End Set
        End Property

        Dim fLocationName As String
        <Size(50)>
        Public Property LocationName() As String
            Get
                Return fLocationName
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationName", fLocationName, value)
            End Set
        End Property

        Dim fLocationAddress As String
        <Size(150)>
        Public Property LocationAddress() As String
            Get
                Return fLocationAddress
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationAddress", fLocationAddress, value)
            End Set
        End Property

        Dim fLocationCity As String
        <Size(150)>
        Public Property LocationCity() As String
            Get
                Return fLocationCity
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationCity", fLocationCity, value)
            End Set
        End Property

        Dim fLocationState As String
        <Size(50)>
        Public Property LocationState() As String
            Get
                Return fLocationState
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationState", fLocationState, value)
            End Set
        End Property

        Dim fLocationZipCode As String
        <Size(50)>
        Public Property LocationZipCode() As String
            Get
                Return fLocationZipCode
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationZipCode", fLocationZipCode, value)
            End Set
        End Property

        Dim fLocationPhone As String
        <Size(50)>
        Public Property LocationPhone() As String
            Get
                Return fLocationPhone
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationPhone", fLocationPhone, value)
            End Set
        End Property

        Dim fLocationFax As String
        <Size(50)>
        Public Property LocationFax() As String
            Get
                Return fLocationFax
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("LocationFax", fLocationFax, value)
            End Set
        End Property

        Dim fCanReceive As Boolean
        Public Property CanReceive As Boolean
            Get
                Return fCanReceive
            End Get
            Set(value As Boolean)
                SetPropertyValue(Of Boolean)("CanReceive", fCanReceive, value)
            End Set
        End Property

        Dim fCanProduce As Boolean
        Public Property CanProduce As Boolean
            Get
                Return fCanProduce
            End Get
            Set(value As Boolean)
                SetPropertyValue(Of Boolean)("CanProduce", fCanProduce, value)
            End Set
        End Property

        Dim fCanShip As Boolean
        Public Property CanShip As Boolean
            Get
                Return fCanShip
            End Get
            Set(value As Boolean)
                SetPropertyValue(Of Boolean)("CanShip", fCanShip, value)
            End Set
        End Property

        Dim fCanWarehouse As Boolean
        Public Property CanWarehouse As Boolean
            Get
                Return fCanWarehouse
            End Get
            Set(value As Boolean)
                SetPropertyValue(Of Boolean)("CanWarehouse", fCanWarehouse, value)
            End Set
        End Property

        Dim fInventoryAvailable As Boolean
        Public Property InventoryAvailable As Boolean
            Get
                Return fInventoryAvailable
            End Get
            Set(value As Boolean)
                SetPropertyValue(Of Boolean)("InventoryAvailable", fInventoryAvailable, value)
            End Set
        End Property

        <Association("Location-Inventory", GetType(LocationInventory)), Aggregated()>
        Public ReadOnly Property LocationsInventory As XPCollection(Of LocationInventory)
            Get
                Return GetCollection(Of LocationInventory)("LocationsInventory")
            End Get
        End Property

        <Association("Location-InventoryByLot", GetType(LocationInventoryByLot)), Aggregated()>
        Public ReadOnly Property LocationsInventoryByLot As XPCollection(Of LocationInventoryByLot)
            Get
                Return GetCollection(Of LocationInventoryByLot)("LocationsInventoryByLot")
            End Get
        End Property

        <Association("Location-TransfersFrom", GetType(LocationTransfers)), Aggregated()>
        Public ReadOnly Property LocationTransfersFrom As XPCollection(Of LocationTransfers)
            Get
                Return GetCollection(Of LocationTransfers)("LocationTransfersFrom")
            End Get
        End Property

        <Association("Location-TransfersTo", GetType(LocationTransfers)), Aggregated()>
        Public ReadOnly Property LocationTransfersTo As XPCollection(Of LocationTransfers)
            Get
                Return GetCollection(Of LocationTransfers)("LocationTransfersTo")
            End Get
        End Property

        <Association("Location-Receivings", GetType(Receiving)), Aggregated()>
        Public ReadOnly Property LocationReceivings As XPCollection(Of Receiving)
            Get
                Return GetCollection(Of Receiving)("LocationReceivings")
            End Get
        End Property

        <Association("Location-Shippings", GetType(Shipping)), Aggregated()>
        Public ReadOnly Property LocationShippings As XPCollection(Of Shipping)
            Get
                Return GetCollection(Of Shipping)("LocationShippings")
            End Get
        End Property

        <Association("Production-Location", GetType(Inventory)), Aggregated()>
        Public ReadOnly Property LocationProduction As XPCollection(Of Inventory)
            Get
                Return GetCollection(Of Inventory)("LocationProduction")
            End Get
        End Property

        Public Sub New()
            MyBase.New()
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
            ' Place here your initialization code.
        End Sub

        Public Shadows Class FieldsClass
            Inherits XPLiteObject.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property Oid() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Oid"))
                End Get
            End Property
            Public ReadOnly Property LocationCode() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationCode"))
                End Get
            End Property
            Public ReadOnly Property LocationName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationName"))
                End Get
            End Property
            Public ReadOnly Property LocationAddress() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationAddress"))
                End Get
            End Property
            Public ReadOnly Property LocationCity() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationCity"))
                End Get
            End Property
            Public ReadOnly Property LocationState() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationState"))
                End Get
            End Property
            Public ReadOnly Property LocationZipCode() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationZipCode"))
                End Get
            End Property
            Public ReadOnly Property LocationPhone() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationPhone"))
                End Get
            End Property
            Public ReadOnly Property LocationFax() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LocationFax"))
                End Get
            End Property
            Public ReadOnly Property CanReceive() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CanReceive"))
                End Get
            End Property
            Public ReadOnly Property CanProduce() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CanProduce"))
                End Get
            End Property
            Public ReadOnly Property CanShip() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CanShip"))
                End Get
            End Property
            Public ReadOnly Property CanWarehouse() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CanWarehouse"))
                End Get
            End Property
            Public ReadOnly Property InventoryAvailable() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("InventoryAvailable"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property

    End Class
End Namespace

File: DXDAL/Locations.cs
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace DXDAL
{
    namespace SPGData
    {
        public class Locations : AuditableBase
        {
            private string fLocationCode;
            [Size(25)]
            public string LocationCode
            {
                get => fLocationCode;
                set => SetPropertyValue("LocationCode", ref fLocationCode, value);
            }

            private string fLocationName;
            [Size(50)]
            public string LocationName
            {
                get => fLocationName;
                set => SetPropertyValue("LocationName", ref fLocationName, value);
            }

            private string fLocationAddress;
            [Size(150)]
            public string LocationAddress
            {
                get => fLocationAddress;
                set => SetPropertyValue("LocationAddress", ref fLocationAddress, value);
            }

            private string fLocationCity;
            [Size(150)]
            public string LocationCity
            {
                get => fLocationCity;
                set => SetPropertyValue("LocationCity", ref fLocationCity, value);
            }

            private string fLocationState;
            [Size(50)]
            public string LocationState
            {
                get => fLocationState;
                set => SetPropertyValue("LocationState", ref fLocationState, value);
            }

            private string fLocationZipCode;
            [Size(50)]
            public string LocationZipCode
            {
                get => fLocationZipCode;
                set => SetPropertyValue("LocationZipCode", ref fLocationZipCode, value);
            }

            private string fLocationPhone;
            [Size(50)]
            public string LocationPhone
            {
                get => fLocationPhone;
                set => SetPropertyValue("LocationPhone", ref fLocationPhone, value);
            }

            private string fLocationFax;
            [Size(50)]
            public string LocationFax
            {
                get => fLocationFax;
                set => SetPropertyValue("LocationFax", ref fLocationFax, value);
            }

            private bool fCanReceive;
            public bool CanReceive
            {
                get => fCanReceive;
                set => SetPropertyValue("CanReceive", ref fCanReceive, value);
            }

            private bool fCanProduce;
            public bool CanProduce
            {
                get => fCanProduce;
                set => SetPropertyValue("CanProduce", ref fCanProduce, value);
            }

            private bool fCanShip;
            public bool CanShip
            {
                get => fCanShip;
                set => SetPropertyValue("CanShip", ref fCanShip, value);
            }

            private bool fCanWarehouse;
            public bool CanWarehouse
            {
                get => fCanWarehouse;
                set => SetPropertyValue("CanWarehouse", ref fCanWarehouse, value);
            }

            private bool fInventoryAvailable;
            public bool InventoryAvailable
            {
                get => fInventoryAvailable;
                set => SetPropertyValue("InventoryAvailable", ref fInventoryAvailable, value);
            }

            [Association("Location-Inventory", typeof(LocationInventory))]
            [Aggregated()]
            public XPCollection<LocationInventory> LocationsInventory => GetCollection<LocationInventory>("LocationsInventory");

            [Association("Location-InventoryByLot", typeof(LocationInventoryByLot))]
            [Aggregated()]
            public XPCollection<LocationInventoryByLot> LocationsInventoryByLot => GetCollection<LocationInventoryByLot>("LocationsInventoryByLot");

            [Association("Location-TransfersFrom", typeof(LocationTransfers))]
            [Aggregated()]
            public XPCollection<LocationTransfers> LocationTransfersFrom => GetCollection<LocationTransfers>("LocationTransfersFrom");

            [Association("Location-TransfersTo", typeof(LocationTransfers))]
            [Aggregated()]
            public XPCollection<LocationTransfers> LocationTransfersTo => GetCollection<LocationTransfers>("LocationTransfersTo");

            [Association("Location-Receivings", typeof(Receiving))]
            [Aggregated()]
            public XPCollection<Receiving> LocationReceivings => GetCollection<Receiving>("LocationReceivings");

            [Association("Location-Shippings", typeof(Shipping))]
            [Aggregated()]
            public XPCollection<Shipping> LocationShippings => GetCollection<Shipping>("LocationShippings");

            [Association("Production-Location", typeof(Inventory))]
            [Aggregated()]
            public XPCollection<Inventory> LocationProduction => GetCollection<Inventory>("LocationProduction");

            public Locations() : base()
            {
            }

            public Locations(Session session) : base(session)
            {
            }

            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty Oid => new OperandProperty(GetNestedName("Oid"));
                public OperandProperty LocationCode => new OperandProperty(GetNestedName("LocationCode"));
                public OperandProperty LocationName => new OperandProperty(GetNestedName("LocationName"));
                public OperandProperty LocationAddress => new OperandProperty(GetNestedName("LocationAddress"));
                public OperandProperty LocationCity => new OperandProperty(GetNestedName("LocationCity"));
                public OperandProperty LocationState => new OperandProperty(GetNestedName("LocationState"));
                public OperandProperty LocationZipCode => new OperandProperty(GetNestedName("LocationZipCode"));
                public OperandProperty LocationPhone => new OperandProperty(GetNestedName("LocationPhone"));
                public OperandProperty LocationFax => new OperandProperty(GetNestedName("LocationFax"));
                public OperandProperty CanReceive => new OperandProperty(GetNestedName("CanReceive"));
                public OperandProperty CanProduce => new OperandProperty(GetNestedName("CanProduce"));
                public OperandProperty CanShip => new OperandProperty(GetNestedName("CanShip"));
                public OperandProperty CanWarehouse => new OperandProperty(GetNestedName("CanWarehouse"));
                public OperandProperty InventoryAvailable => new OperandProperty(GetNestedName("InventoryAvailable"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
